File: Backend/app/Models/teacher.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Illuminate\Notifications\Notifiable;
use Laravel\Sanctum\HasApiTokens;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Teacher extends Authenticatable
{
    use HasApiTokens, HasFactory, Notifiable;
    
    protected $fillable = [
        "username",
        "fullname",
        "email", 
        "password",
        "specialist",
        "picture",
        "picture_mime",
    ];

    protected $hidden = [
        'password',
        'remember_token',
        'picture',
        'picture_mime',
    ];

    protected $appends = ['picture_url'];

    protected $primaryKey = 'teacher_id';
    public $incrementing = true;
    protected $keyType = 'integer';

    // Route model binding key
    public function getRouteKeyName()
    {
        return 'teacher_id';
    }

    public function classes(): HasMany
    {
        return $this->hasMany(Classes::class, 'teacher_id', 'teacher_id');
    }

    public function charadesThemes(): HasMany
    {
        return $this->hasMany(CharadesThemes::class, 'teacher_id', 'teacher_id');
    }

    public function gameSessions(): HasMany
    {
        return $this->hasMany(GameSession::class, 'teacher_id', 'teacher_id');
    }

    /**
     * Get the URL for the teacher's picture.
     */
    public function getPictureUrlAttribute(): ?string
    {
        return url('api/teachers/' . $this->teacher_id . '/picture');
    }
}

File: frontend/lib/core/models/teacher.dart
class Teacher {
  final int teacherId;
  final String username;
  final String fullname;
  final String email;
  final String? specialist;
  final String? pictureUrl;

  Teacher({
    required this.teacherId,
    required this.username,
    required this.fullname,
    required this.email,
    this.specialist,
    this.pictureUrl,
  });

  factory Teacher.fromJson(Map<String, dynamic> json) {
    return Teacher(
      teacherId: json['teacher_id'],
      username: json['username'],
      fullname: json['fullname'] ?? json['username'],
      email: json['email'],
      specialist: json['specialist'],
      pictureUrl: json['picture_url'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'teacher_id': teacherId,
      'username': username,
      'fullname': fullname,
      'email': email,
      'specialist': specialist,
      'picture_url': pictureUrl,
    };
  }
}

class LoginResponse {
  final bool success;
  final String message;
  final Teacher teacher;
  final String token;

  LoginResponse({
    required this.success,
    required this.message,
    required this.teacher,
    required this.token,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      success: json['success'],
      message: json['message'],
      teacher: Teacher.fromJson(json['teacher']),
      token: json['token'],
    );
  }
}